lib/core/scheduler.rb

```diff
@@ -179,12 +179,15 @@ def schedule_triggered?(scheduling_settings, identifier)
         return false
       end
 
-      next_trigger_time = cron_parser.next_time(Time.now)
-
+      next_trigger_time = cron_parser.next_time(time_at_poll_start)
       # Sync if next trigger happens before the next poll
-      if next_trigger_time <= Time.now + @poll_interval
+      poll_window = time_at_poll_start + @poll_interval
+      if next_trigger_time <= poll_window
         Utility::Logger.info("#{identifier.capitalize} sync is triggered by cron schedule #{current_schedule}.")
         return true
+      else
+        # log that a sync was not triggered, share the next trigger time and when poll interval was meant to end
+        Utility::Logger.debug("Sync for #{identifier.capitalize} not triggered, as #{next_trigger_time} occurs after the current interval: #{poll_window}.")
```

Comment:
Nit/discussion: I wonder which would be more valuable to the user, the `poll_window` value, or a description of the `time_at_poll_start` and their configured `poll_interval` 🤔 
Maybe all of this information can be included, if the log isn't getting too long. What are your thoughts?

```suggestion
        Utility::Logger.debug("Sync for #{identifier.capitalize} not triggered, as #{next_trigger_time} occurs after the current poll window: #{poll_window}. Poll window began at #{time_at_poll_start}, poll interval is #{@poll_interval}" seconds.)
```

Reply:
That's a really good idea - I agree adding the value of `poll_interval` makes a lot of sense here as well. I'll play around with the verbiage to see how we can incorporate the info we want in a concise manner.

If it gets too wordy I could possibly break this out into 2 debug lines maybe?

Reply:
So this morning when I responded to this comment my brain didn't compute the fact that you had already suggested a change @navarone-feekery  - sorry for the brainfart! I've incorporated your wording as it's far better than what I was going to come up with 🤣

Reply:
I don't think anyone is picky about whether or not you click "accept changes" or you manually add the changes. I'm not chasing github points or anything, it's just a nice format to suggest a change. How the change goes in is inconsequential, so it's not something to worry about :)